tests/build/build_templates.go

```diff
@@ -353,6 +347,116 @@ var _ = framework.BuildSuiteDescribe("Build templates E2E test", Label("build",
 					}
 				})
 			})
+
+			Context("build-definitions ec pipelines", func() {
+				ecPipelines := []string{
+					"pipelines/enterprise-contract.yaml",
+					"pipelines/enterprise-contract-everything.yaml",
+					"pipelines/enterprise-contract-redhat.yaml",
+					"pipelines/enterprise-contract-slsa1.yaml",
+					"pipelines/enterprise-contract-slsa2.yaml",
+					"pipelines/enterprise-contract-slsa3.yaml",
+				}
+
+				var gitRevision, gitURL, imageWithDigest string
+				var kube tekton.KubeController
+
+				defaultGHOrg := "redhat-appstudio"
+				defaultGHRepo := "build-definitions"
+				defaultGitURL := fmt.Sprintf("https://github.com/%s/%s", defaultGHOrg, defaultGHRepo)
+				defaultGitRevision := "main"
+
+				BeforeAll(func() {
+					// If we are testing the changes from a pull request, APP_SUFFIX may contain the
+					// pull request ID. If it looks like an ID, then fetch information about the pull
+					// request and use it to determine which git URL and revision to use for the EC
+					// pipelines. NOTE: This is a workaround until Pipeline as Code supports passing
+					// the source repo URL: https://issues.redhat.com/browse/SRVKP-3427. Once that's
+					// implemented, remove the APP_SUFFIX support below and simply rely on the other
+					// environment variables to set the git revision and URL directly.
+					appSuffix := os.Getenv("APP_SUFFIX")
+					if pullRequestID, err := strconv.ParseInt(appSuffix, 10, 64); err == nil {
+						gh, err := github.NewGithubClient(utils.GetEnv(constants.GITHUB_TOKEN_ENV, ""), defaultGHOrg)
+						Expect(err).NotTo(HaveOccurred())
+						pullRequest, err := gh.GetPullRequest(defaultGHRepo, int(pullRequestID))
+						Expect(err).NotTo(HaveOccurred())
+						gitURL = *pullRequest.Head.Repo.CloneURL
+						gitRevision = *pullRequest.Head.Ref
+						return
+					}
+
+					gitRevision = utils.GetEnv(constants.EC_PIPELINES_REPO_REVISION_ENV, defaultGitRevision)
+					gitURL = utils.GetEnv(constants.EC_PIPELINES_REPO_URL_ENV, defaultGitURL)
```

Comment:
Should this be in a `else` of the `if` above? Kinda hard for me to see the intent behind this since it'll overwrite `gitURL` and `gitRevision` if set in the `if-true` above